localstack/services/firehose/firehose_api.py

```diff
@@ -149,15 +149,35 @@ def put_record(stream_name: str, record: Dict) -> Dict:
     return put_records(stream_name, [record])
 
 
-def put_records(stream_name: str, records: List[Dict]) -> Dict:
+def put_records_to_s3_bucket(
+    stream_name: str, records: List[Dict], s3_configuration: Dict[str, Union[str, Dict]]
+):
+    bucket = s3_bucket_name(s3_configuration["BucketARN"])
+    prefix = s3_configuration.get("Prefix", "")
+
+    s3 = connect_to_resource("s3")
+    batched_data = b"".join([base64.b64decode(r.get("Data") or r["data"]) for r in records])
```

Comment:
nit: I think `Data` may potentially contain an empty string (at least kinesis-mock allows to push records with empty data), which could then result in `KeyError` if it falls back to `r["data"]`.

We could do something like:
```
batched_data = b"".join([base64.b64decode(r.get("Data", r["data"])) for r in records])
```
?